src/components/timeline/timeline.tsx

```diff
@@ -8,26 +8,57 @@ import { Slider } from "@mui/material";
 import css from "./timeline.scss";
 
 const TICK_COUNT = 16;
+const LOADING_DELAY = 100; // ms
 
 export const Timeline: React.FC = observer(function WrappedComponent() {
-  const { simulation } = useStores();
+  const { simulation, snapshotsManager } = useStores();
+  const timeoutId = useRef(0);
   const [val, setVal] = useState(simulation.timeInYears);
   const [disabled, setDisabled] = useState(true);
+  const [timeProgress, setTimeProgress] = useState("0%");
 
   const tickDiff = simulation.config.simulationEndYear / TICK_COUNT;
   const ticks = Array.from({ length: TICK_COUNT + 1 }, (_, i) => i * tickDiff);
-  const timeProgress = `${simulation.timeInYears / simulation.config.simulationEndYear * 100}%`;
   const marks = ticks.map((tick) => ({ value: tick, label: tick }));
   useEffect(() => {
     setDisabled(!simulation.simulationStarted || (simulation.simulationRunning && !simulation.simulationEnded));
   }, [simulation.simulationStarted, simulation.simulationRunning, simulation.simulationEnded]);
 
   useLayoutEffect(() => {
-    setVal(simulation.timeInYears);
-  }, [simulation.timeInYears]);
+    if (simulation.simulationRunning && snapshotsManager.maxYear > val) {
+      snapshotsManager.restoreLastSnapshot();
+      simulation.start(); //when we restoreLastSnapshot, we need to start the simulation again
```

Comment:
When this block is executed? It doesn't seem we ever (re)starting the simulation when we use the new timeline slider. So, I'm wondering if it's not a dead code?